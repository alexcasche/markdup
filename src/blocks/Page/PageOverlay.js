import styled from 'styled-components'

const PageOverlay = styled.div`
  position: fixed;
  top: 50px;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: ${props => props.theme.colors.black};
  opacity: 0.5;
  z-index: 9;
`

export default PageOverlay
