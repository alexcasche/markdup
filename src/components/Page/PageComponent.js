import React from 'react'
import { Page } from 'blocks'
import { Banner, Navbar, Reflex } from 'components'
import { PropTypes as t } from 'prop-types'

const PageComponent = props => {
  const { showSettings, showSidebar, actions } = props
  const pageModifiers = () => {
    return [...(showSidebar ? ['sidebar'] : [])]
  }
  return (
    <Page className="Page" modifiers={pageModifiers()}>
      <Navbar />
      <Banner />
      <Reflex />
      {showSettings || showSidebar ? (
        <Page.Overlay
          onClick={() =>
            actions.updateLayout({ showSettings: false, showSidebar: false })
          }
        />
      ) : null}
    </Page>
  )
}

PageComponent.propTypes = {
  showSettings: t.bool.isRequired,
  showSidebar: t.bool.isRequired,
  actions: t.shape({
    updateLayout: t.func.isRequired,
  }).isRequired,
}

export default PageComponent
