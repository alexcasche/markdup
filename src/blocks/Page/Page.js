import styled from 'styled-components'
import { applyStyleModifiers } from 'styled-components-modifiers'
import PageOverlay from './PageOverlay'

const MODIFIERS = {
  sidebar: () => `
    transform: translateX(300px);
  `,
}

const Page = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #fafafa;
  height: 100%;
  overflow: hidden;
  transition: transform 0.35s ease-in-out;
  z-index: 1;
  ${applyStyleModifiers(MODIFIERS)};
`

Page.Overlay = PageOverlay

export default Page
