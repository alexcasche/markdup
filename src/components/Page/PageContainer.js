import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getLayoutVal } from 'selectors'
import { updateLayout } from 'actions'
import PageComponent from './PageComponent'

export const PageContainer = props => {
  return <PageComponent {...props} />
}

function mapStateToProps(state) {
  return {
    showSettings: getLayoutVal(state, 'showSettings'),
    showSidebar: getLayoutVal(state, 'showSidebar'),
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: {
      updateLayout: bindActionCreators(updateLayout, dispatch),
    },
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PageContainer)
