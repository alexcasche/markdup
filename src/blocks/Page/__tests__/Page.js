import Page from '../Page'

describe('Page Block', () => {
  it('renders correctly', () => {
    const tree = render(<Page />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds sidebar modifier', () => {
    const tree = render(<Page modifiers={['sidebar']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('transform', 'translateX(300px)')
  })
})
