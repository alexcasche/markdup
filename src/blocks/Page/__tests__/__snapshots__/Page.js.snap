// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Block adds sidebar modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fafafa;
  height: 100%;
  overflow: hidden;
  -webkit-transition: -webkit-transform 0.35s ease-in-out;
  -webkit-transition: transform 0.35s ease-in-out;
  transition: transform 0.35s ease-in-out;
  z-index: 1;
  -webkit-transform: translateX(300px);
  -ms-transform: translateX(300px);
  transform: translateX(300px);
}

<div
  className="c0"
/>
`;

exports[`Page Block renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fafafa;
  height: 100%;
  overflow: hidden;
  -webkit-transition: -webkit-transform 0.35s ease-in-out;
  -webkit-transition: transform 0.35s ease-in-out;
  transition: transform 0.35s ease-in-out;
  z-index: 1;
}

<div
  className="c0"
/>
`;
